tools/clang/unittests/HLSL/ShaderOpTest.h

```diff
@@ -167,6 +167,13 @@ class ShaderOpRootValue {
   UINT    Index;      // Explicit index in root table.
 };
 
+// Use this class to represent a render target and its viewport.
+class ShaderOpRenderTarget {
+public:
+  LPCSTR             Name;        // Render target name
+  std::vector<BYTE>  InitBytes;   // Byte payload for initialization of D3D12_VIEWPORT
```

Comment:
Why `std::vector<BYTE> InitBytes;` instead of `D3D12_VIEWPORT Viewport;` here?  Is it just to know if it wasn't written?  It's just that InitBytes here might imply initializing the render target resource with these bytes.  If the viewport type is used, that would mean copying from a local `std::vector<BYTE>` output to the structure in `ParseRenderTargets()` when the size is correct or something like that.  Perhaps a zero-initialized viewport could trigger default behavior, or a separate boolean for custom Viewport provided?

Of course using ReadAttrBytes for this does seem like a bit of a shortcut born of not wanting to type a bit more XML to describe the viewport properties in a clearer, more verbose way, like the other structures have been mapped.  So perhaps the right thing would just be to expose the structure rather than relying on the raw bytes path.

Reply:
I have added support for <Viewport> element under <RenderTarget> with the individual viewport values as XML attributes.